### PR TITLE
feat: script extensions sorter checks load order

### PR DESCRIPTION
**Pull request by KANAjetzt**
Cherry-picked #357 

Original description:
> the script extensions sorter will check load order to sort extensions of the same vanilla script
> 
> inheritance is still the first way of sorting, this only sorts conflicting extensions of the same vanilla script

### CODE REVIEW

File: addons/mod_loader/internal/path.gd [hunk 1]
```diff
@@ -180,4 +180,26 @@ static func get_path_to_mod_configs_dir(mod_id: String) -> String:
 static func get_path_to_mod_config_file(mod_id: String, config_name: String) -> String:
 	var mod_config_dir := get_path_to_mod_configs_dir(mod_id)
 
+
 	return mod_config_dir.path_join(config_name + ".json")
+
+
+# Returns the mod directory name ("some-mod") from a given path (e.g. "res://mods-unpacked/some-mod/extensions/extension.gd")
+static func get_mod_dir(path: String) -> String:
+	var initial = ModLoaderStore.UNPACKED_DIR
```
Comment (Qubus0) [inline addons/mod_loader/internal/path.gd:189]: types 👀

File: addons/mod_loader/internal/script_extension.gd [hunk 2]
```diff
@@ -32,8 +32,12 @@ static func handle_script_extensions() -> void:
 # a script extending script B if A is an ancestor of B.
 class InheritanceSorting:
 	var stack_cache := {}
+	# This dictionary's keys are mod_ids and it stores the corresponding position in the load_order
+	var load_order := {}
+	var unpacked_dir = _ModLoaderPath.get_unpacked_mods_dir_path()
```
Comment (Qubus0) [inline addons/mod_loader/internal/script_extension.gd:37]: seems unused?

**Review by Qubus0 — COMMENTED** [review]
right. this one fixes one issue where script extensions were loaded based on filename even if the mod dependencies said otherwise 👍

File: addons/mod_loader/internal/path.gd (hunk 1 above)
Comment (KANAjetzt) [reply addons/mod_loader/internal/path.gd]: ty - fixed [a697c3d](https://github.com/GodotModding/godot-mod-loader/pull/382/commits/a697c3dbf4efc1305d9f3505fd29c2531598e6d3)

File: addons/mod_loader/internal/script_extension.gd (hunk 2 above)
Comment (KANAjetzt) [reply addons/mod_loader/internal/script_extension.gd]: True, good catch 🎣 
I opened an issue, so we don't forget to removed it on the 3.x branch.

[554b75e](https://github.com/GodotModding/godot-mod-loader/pull/382/commits/554b75ed6aede4765e607d6e0f19201a632c163e)
#389

**Review by Qubus0 — APPROVED** [review]

**Pull request merged**